File: load_era.R
## this works, but make sure you copy .ecmwfapirc to your home directory (check 3.2 here: https://dominicroye.github.io/en/2018/access-to-climate-reanalysis-data-from-r/#connection-and-download-with-the-ecmwf-api):
Sys.setenv(RETICULATE_PYTHON='/home/delgado/local/miniconda3/envs/ecmwf/bin/python')
library(reticulate)
py_config()
ecmwf <- import('ecmwfapi')

server = ecmwf$ECMWFDataServer(verbose=TRUE)
query = r_to_py(list(
  class='ei',
  dataset='interim',
  date='2018-01-01/to/2018-01-02',
  expver='1',
  grid='0.125/0.125',
  levtype='sfc',
  param='167.128', # air temperature (2m), check parameter db in https://apps.ecmwf.int/codes/grib/param-db
  area='31/4/30/5', # N/W/S/E
  step='0',
  stream='oper',
  time='00/12',
  type='an',
  format='netcdf',
  target='era_test.nc'
  ))


server$retrieve(query)

v=get(load('./data/vartable.RData'))

library(scraping)


my_dataset='GPCC'
my_var='precipitation'

    require(dplyr)


vartable=tibble(variable='2 metre temperature',varname='165.128',dataset='era-interim') %>%
  bind_rows(v,.) %>% mutate(dataset=tolower(dataset))

  devtools::use_data(vartable,pkg='.')
  usethis::use_data(vartable,pkg='scraping')

  vartable




getwd()

File: load_era.py
#!/usr/bin/env python
from ecmwfapi import ECMWFDataServer

server = ECMWFDataServer()

server.retrieve({
    'stream'    : "oper",
    'levtype'   : "sfc",
    'param'     : "165.128/166.128/167.128",
    'dataset'   : "interim",
    'step'      : "0",
    'grid'      : "0.75/0.75",
    'time'      : "00/06/12/18",
    'date'      : "2014-07-01/to/2014-07-31",
    'type'      : "an",
    'class'     : "ei",
    'area'      : "73.5/-27/33/45",
    'format'    : "netcdf",
    'target'    : "interim_2014-07-01to2014-07-31_00061218.nc"
 })
